lib/dradis/plugins/wpscan/importer.rb

```diff
@@ -12,16 +12,140 @@ def import(params={})
       # format.
-      unless data.key?("scan_info")
-        logger.error "ERROR: no 'scan_info' field present in the provided "\
-                     "data. Are you sure you uploaded a Brakeman file?"
+      if data['target_url'].nil?
+        logger.error "ERROR: No 'target_url' field present in the provided "\
+                     "JSON data. Are you sure you uploaded a WPScan JSON output file?"
         exit(-1)
       end
 
-      # Parse the `data` structure and call template_service and
-      # content_service to create data in the project.
+      # Parse scan info data and make more human readable.
+      data['wpscan_version']    = data.dig('banner', 'version')
+      data['start_time']        = DateTime.strptime(data['start_time'].to_s,'%s')
+      data['elapsed']           = "#{data["elapsed"]} seconds"
+      data['wordpress_version'] = data.dig('version', 'number')   if data['version']
+      data['plugins_string']    = data['plugins'].keys.join("\n") if data['plugins']
+      data['themes_string']     = data['themes'].keys.join("\n")  if data['themes']
+      data['users']             = data['users'].keys.join("\n")   if data['users']
+
+      scan_info = template_service.process_template(template: 'scan_info', data: data)
+      content_service.create_note text: scan_info
+
+      # Parse vulnerability data and make more human readable.
+      # NOTE: You need an API token for the WPVulnDB vulnerability data.
+      vulnerabilities = []
+
+      # WordPress Vulnerabilities
+      if data['version'] && data['version']['status'] == 'insecure'
+        data['version']['vulnerabilities'].each do |vulnerability_data|
+          vulnerabilities << add_vulnerability( vulnerability_data )
+        end
+      end
+
+      # Plugin Vulnerabilities
+      if data['plugins']
+        data['plugins'].each do |key, plugin|
+          if plugin['vulnerabilities']
+            plugin['vulnerabilities'].each do |vulnerability_data|
+              vulnerabilities << add_vulnerability( vulnerability_data )
+            end
+          end
+        end
+      end
+
+      # Theme Vulnerabilities
+      if data['themes']
+        data['themes'].each do |key, theme|
+          if theme['vulnerabilities']
+            theme['vulnerabilities'].each do |vulnerability_data|
+              vulnerabilities << add_vulnerability( vulnerability_data )
+            end
+          end
+        end
+      end
+
+      # Add vulnerabilities from WPVulnDB to Dradis
+      vulnerabilities.each do |vulnerability|
+        logger.info { "Adding vulnerability: #{vulnerability['title']}" }
+
+        vulnerability_template = template_service.process_template(template: 'vulnerability', data: vulnerability)
+        content_service.create_issue(text: vulnerability_template, id: vulnerability['wpvulndb_id'])
+      end
+
+      # Add bespoke/config vulnerabilities to Dradis
+      #
+      # TODO: Would be better to add the URL & passwords as evidence.
```

Comment:
An idea that may improve on this:

at the beginning of the import, you may create a node with the sacaned target_url as label (similar to what it is going on here: https://github.com/dradis/dradis-acunetix/blob/master/lib/dradis/plugins/acunetix/importer.rb#L36)

Then you can create the issues as you are doing, but you can also create evidence  to show that those issues are happening in that node (target_url).

This way of later we import another wpscan file with another target_url, existing issues won't be duplicated, but the evidence will be created for the new target_url node.

Reply:
Thanks for your feedback!

I'll implement the changes tonight/tomorrow sometime.

Reply:
Note that this suggestion only makes sense if we can find the evidence that relates each issue (vulnerability) to the node (target_url).